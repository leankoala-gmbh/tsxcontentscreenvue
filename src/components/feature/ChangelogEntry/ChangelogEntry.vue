<script lang="ts" setup>
import { IChangelogList } from '@/types/general'
import { marked } from 'marked'

const props = defineProps({
  entry: {
    type: Object as () => IChangelogList,
    required: true
  }
})

const { translate, getLanguage } = useTranslator()

const markdownBody = computed(() => {
  if (!props.entry.body) return null
  return marked(props.entry.body)
})
</script>

<template>
  <div>
    <div class="flex items-center gap-3 mb-3">
      <div class="rounded px-2 py-1" :class="[`changelogColors--${entry.type}`]">
        {{ translate(entry.type) }}
      </div>
      <div class="text-slate-500">
        {{ new Date(entry.changelogCreatedAt).toLocaleDateString(getLanguage()) }}
      </div>
    </div>
    <h2 class="text-xl font-medium mb-4">
      {{ entry.title }}
    </h2>
    <div
      v-if="markdownBody"
      class="richTextContent richTextContent--changelog"
      v-html="markdownBody"
    />
  </div>
</template>
