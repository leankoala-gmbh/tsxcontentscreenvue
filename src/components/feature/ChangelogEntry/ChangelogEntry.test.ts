import { describe, it, expect } from 'vitest'
import { mount } from '@vue/test-utils'
import ChangelogEntry from './ChangelogEntry.vue'
import changeLogsResponse from '../../../mockData/changelog.mock.json'

describe('ChangelogEntry.vue', () => {
  it('should match snapshot', () => {
    const wrapper = mount(ChangelogEntry, {
      props: {
        entry: changeLogsResponse[0]
      }
    })
    expect(wrapper.html()).toMatchSnapshot()
  })
})
