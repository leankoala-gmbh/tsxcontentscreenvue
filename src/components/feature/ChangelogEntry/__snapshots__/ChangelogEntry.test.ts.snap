// Vitest Snapshot v1

exports[`ChangelogEntry.vue > should match snapshot 1`] = `
"<div>
  <div class=\\"flex items-center gap-3 mb-3\\">
    <div class=\\"rounded px-2 py-1 changelogColors--improved\\">Improved</div>
    <div class=\\"text-slate-500\\">1/10/2023</div>
  </div>
  <h2 class=\\"text-xl font-medium mb-4\\">The best one</h2>
  <div class=\\"richTextContent richTextContent--changelog\\">
    <h3 id=\\"sascha-is-awesome\\">Sascha is awesome!</h3>
    <blockquote>
      <p>By Sascha himself
        👍 This is Success</p>
    </blockquote>
    <blockquote>
      <p>📘 Info</p>
    </blockquote>
    <blockquote>
      <p>🚧 Warning</p>
    </blockquote>
    <blockquote>
      <p>❗️ Errors are the best</p>
      <p>They deserve a description</p>
    </blockquote>
    <ul>
      <li>
        <p>Bullet</p>
      </li>
      <li>
        <p>Another one</p>
      </li>
      <li>
        <p>And another one</p>
      </li>
      <li>
        <p><input checked=\\"\\" disabled=\\"\\" type=\\"checkbox\\">
          We are good</p>
      </li>
      <li>
        <p><input checked=\\"\\" disabled=\\"\\" type=\\"checkbox\\">
          As usual</p>
      </li>
    </ul>
    <hr>
    <p>Faster then an eagle</p>
    <p><img src=\\"https://files.readme.io/53726f5-eagle.jpeg\\" alt=\\"\\"></p>
    <h1 id=\\"big-improvements\\">Big improvements!</h1>
    <pre><code>/*
 * Dear Maintainer
 *
 * Once you are done trying to ‘optimize’ this routine,
 * and you have realized what a terrible mistake that was,
 * please increment the following counter as a warning
 * to the next guy.
 *
 * total_hours_wasted_here = 73
 */
</code></pre>
    <p>Watch the video</p>
    <p>[block:embed]
      {
      \\"url\\": \\"\\",
      \\"provider\\": \\"\\",
      \\"href\\": \\"\\",
      \\"typeOfEmbed\\": \\"youtube\\"
      }
      [/block]</p>
  </div>
</div>"
`;
